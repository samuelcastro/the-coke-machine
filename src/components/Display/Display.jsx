import React from 'react';
import PropTypes from 'prop-types';

const Display = ({value}) => {
  return (
    <section>{value}</section>
  )
};

Display.propTypes = {
  value: PropTypes.any.isRequired
};

export default Display;
