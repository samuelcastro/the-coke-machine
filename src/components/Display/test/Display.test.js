import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../Display';

describe('Rendering Display', () => {
  const mockedValue = 'Welcome'

  it('should render <Display /> correctly', () => {
    const tree = renderer
      .create(<Display value={mockedValue} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a diplay value', () => {
    const wrapper = shallow(<Display value={mockedValue}/>);

    expect(wrapper.find('section').text()).toEqual(mockedValue);
  });
});
