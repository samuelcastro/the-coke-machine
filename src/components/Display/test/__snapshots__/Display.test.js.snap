// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Display should render <Display /> correctly 1`] = `
<section>
  Welcome
</section>
`;
